src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -1341,6 +1342,76 @@ public void testContainerWithCpuQuota() throws Exception {
   }
 
   @Test
+  public void testEventStream() throws Exception {
+    sut.pull("busybox");
+    EventStream eventStream = sut.events();
+    final ContainerConfig config = ContainerConfig.builder()
+            .image("busybox")
+            .build();
+    final ContainerCreation container = sut.createContainer(config, randomName());
+    sut.startContainer(container.id());
+
+    final Event createEvent = eventStream.next();
+    assertThat(createEvent.status(), equalTo("create"));
+    assertThat(createEvent.id(), equalTo(container.id()));
+    assertThat(createEvent.from(), startsWith("busybox:"));
+    assertThat(createEvent.time(), notNullValue());
+
+    final Event startEvent = eventStream.next();
+    assertThat(startEvent.status(), equalTo("start"));
+    assertThat(startEvent.id(), equalTo(container.id()));
+    assertThat(startEvent.from(), startsWith("busybox:"));
+    assertThat(startEvent.time(), notNullValue());
+
+    eventStream.close();
+  }
+
+  @Test
+  public void testEventStreamWithSinceTime() throws Exception {
+    Thread.sleep(1000); // ensure we push to the next second
+                        // so we don't get events from the last test
```

Comment:
@mattnworb the docker daemon doesn't change between tests. The problem here is that if the first test runs at time 1, and the second test starts running at time 1.9, they share the same unix timestamp (in seconds). Then, when the second test runs, it requests events starting from time t=1, and receives events that the daemon generated during the first test's run.

In an ideal world each test would run with an isolated daemon to avoid this problem, but I see no way to make that feasible.